platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -1242,433 +720,501 @@ public void start() {
+                components.add(PlatformConstructor.parallelExecutor(threadManager));
+        final ShadowGraphSynchronizer shadowgraphSynchronizer = new ShadowGraphSynchronizer(
+                shadowGraph,
                 getAddressBook().getSize(),
                 syncMetrics,
-                consensusRef::get,
+                this::getConsensus,
                 eventTaskCreator::syncDone,
                 eventTaskCreator::addEvent,
                 syncManager,
                 shadowgraphExecutor,
                 true,
                 () -> {});
 
-        final Runnable stopGossip = settings.getChatter().isChatterUsed()
-                ? chatterCore::stopChatter
-                // wait and acquire all sync ongoing locks and release them immediately
-                // this will ensure any ongoing sync are finished before we start reconnect
-                // no new sync will start because we have a fallen behind status
-                : getSimultaneousSyncThrottle()::waitForAllSyncsToFinish;
-        reconnectHelper = new ReconnectHelper(
-                stopGossip,
-                clearAllPipelines,
-                getSwirldStateManager()::getConsensusState,
-                stateManagementComponent::getLastCompleteRound,
-                new ReconnectLearnerThrottle(selfId, settings.getReconnect()),
-                this::loadReconnectState,
-                new ReconnectLearnerFactory(
-                        threadManager, initialAddressBook, settings.getReconnect(), reconnectMetrics));
+        this.platformMetrics = new PlatformMetrics(
+                platformContext.getMetrics(),
+                selfId,
+                freezeManager,
+                startUpEventFrozenManager,
+                syncManager,
+                eventMapper,
+                swirldStateManager,
+                criticalQuorum,
+                initialAddressBook,
+                preConsensusEventHandler,
+                consensusRoundHandler,
+                simultaneousSyncThrottle,
+                eventStreamManager,
+                stateManagementComponent);
+        platformContext.getMetrics().addUpdater(platformMetrics::update);
+
         if (settings.getChatter().isChatterUsed()) {
-            reconnectController.set(new ReconnectController(threadManager, reconnectHelper, chatterCore::startChatter));
-            startChatterNetwork();
+            gossipNetwork = components.add(new ChatterNetwork(
+                    platformContext,
+                    threadManager,
+                    crypto,
+                    settings,
+                    initialAddressBook,
+                    selfId,
+                    appVersion,
+                    topology,
+                    this,
+                    time,
+                    networkMetrics,
+                    syncMetrics,
+                    eventTaskCreator,
+                    emergencyRecoveryManager,
+                    shadowGraph,
```

Comment:
All this code shows to me that we need to work on decoupling and inversion-of-control on many places.